File: matrix.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        if matrix == None or len(matrix) == 0:
            pass
        elif len(matrix) == 1 and len(matrix[0]) == 1:
            pass
        else:
            rows_with_0 = [False]*len(matrix)
            cols_with_0 = [False]*len(matrix[0])
            for rows in range(len(matrix)):
                for columns in range(len(matrix[0])):
                    if matrix[rows][columns] == 0:
                        rows_with_0[rows] = True
                        cols_with_0[columns] = True

            for rows in range(len(matrix)):
                for columns in range(len(matrix[0])):
                    if rows_with_0[rows] or cols_with_0[columns]:
                        matrix[rows][columns] = 0

File: javascript_solutions/matrix.js
var setZeroes = function(matrix) {
    let isRow = false
    let isColumn = false
    
for (let i = 0; i < matrix.length; i++) {
  for (let j = 0; j < matrix[i].length; j++) {
    if (matrix[i][j] === 0) {
      if (j === 0) {
        isColumn = true
      }
      if (i === 0) {
        isRow = true
      }
      matrix[i][0] = 0
      matrix[0][j] = 0
    }
  }
}

for (let i = 1; i < matrix.length; i++) {
  for (let j = 1; j < matrix[i].length; j++) {
    if (matrix[0][j] === 0 || matrix[i][0] === 0) {
      matrix[i][j] = 0
    }
  }
}
if (isRow) {
  for (let i = 0; i < matrix[0].length; i++) {
    matrix[0][i] = 0
  }
}
if (isColumn) {
  for (let i = 0; i < matrix.length; i++) {
    matrix[i][0] = 0
  }
}
return matrix
  
  
};